toolkit/tools/internal/miscdebug/miscdebug.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package miscdebug
+
+import (
+	"bufio"
+	"os"
+	"time"
+
+	"microsoft.com/pkggen/internal/logger"
+)
+
+func ic(i *int) {
+	*i++
+}
+
+// WaitForDebugger busy loops until manually broken out of.
+// Useful for breaking in with a debugger when running privileged code.
+func WaitForDebugger(tag string) {
+	i := 1
+	logger.Log.Errorf("Freezing at %s for debugger", tag)
+	logger.Log.Errorf("Use 'break ic', then 'c' to jump to the busy loop")
+
+	for i != 0 {
+		logger.Log.Errorf("Waiting for debugger %d, once broken in run `so` to step out, `set i=0`, then `c`", i)
+		ic(&i)
```

Comment:
Is it simply `i++`? In that case I'd get rid of the helper function.

Reply:
Go will optimize too much if you use i++, and you can't break out anymore. This is some really hacky debugging garbage, best not to think too deeply on it :P

---
In reply to: [552713370](https://github.com/microsoft/CBL-Mariner/pull/483#discussion_r552713370) [](ancestors = 552713370)

Reply:
In that case it would be great to have a comment here mentioning that this function is left for debugging purposes.